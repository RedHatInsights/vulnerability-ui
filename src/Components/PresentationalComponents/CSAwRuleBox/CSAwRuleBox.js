import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Stack,
    StackItem,
    Text,
    TextContent,
    TextVariants,
    Tooltip,
    SplitItem,
    Split,
    Grid,
    GridItem,
    Card,
    CardBody
} from '@patternfly/react-core';
import { Link } from 'react-router-dom';
import CheckCircleIcon from '@patternfly/react-icons/dist/js/icons/check-circle-icon';
import OutlinedQuestionCircleIcon from '@patternfly/react-icons/dist/js/icons/outlined-question-circle-icon';
import ExternalLinkAltIcon from '@patternfly/react-icons/dist/js/icons/external-link-alt-icon';
import { Reboot } from '@redhat-cloud-services/frontend-components/components/Reboot';
import { FormattedMessage } from 'react-intl';

import Label from '../Snippets/Label';
import messages from '../../../Messages';
import { RISK_OF_CHANGE_TOOLTIP, RISK_OF_CHANGE_LABEL, RH_KB_URL, CVES_PATH } from '../../../Helpers/constants';
import TruncatedText from '../Snippets/TruncatedText';
import { CSAwIcon } from '../../PresentationalComponents/CSAwIcon/CSAwIcon';
import './CSAwRuleBox.scss';

const CSAwRuleBox = ({ rules, synopsis, fetchAffectedSystems }) => {
    const handleExposedSystemFilter = (ruleId) => {
        const params = { security_rule: ruleId };
        fetchAffectedSystems(synopsis, params);
    };

    return <Fragment>
        {rules && rules.map(rule => (
            <Card className="card-box-shadow" key={rule.rule_id}>
                <CardBody>
                    <Grid gutter="md">
                        <GridItem span={12}>
                            <TextContent className={'icon-with-label'}>
                                <Text component={TextVariants.h3}>
                                    <Tooltip content={<FormattedMessage {...messages.rulesIconTooltip} />}>
                                        <CSAwIcon />
                                    </Tooltip>
                                    {rule.description}
                                </Text>
                            </TextContent>
                        </GridItem>
                        <GridItem span={8}>
                            <Stack gutter="md">
                                <StackItem>
                                    <TextContent>
                                        <TruncatedText content={rule.summary}/>
                                    </TextContent>
                                </StackItem>

                                <StackItem>
                                    <TextContent>
                                        <Text component={TextVariants.h4}> <FormattedMessage {...messages.associatedCves}/></Text>
                                        <Text component={TextVariants.p}>
                                            {synopsis} (<FormattedMessage {...messages.current}/>)
                                            {
                                                rule.associated_cves
                                                .filter(cve => cve !== synopsis)
                                                .map((cve, _i) =>
                                                    <a
                                                        className="associated-cve-link"
                                                        key={_i}
                                                        href={`${CVES_PATH}/${cve}`}
                                                    >
                                                        {cve}
                                                    </a>
                                                )
                                                .reduce((prev, curr) => [prev, ', ', curr], [''])
                                            }
                                        </Text>
                                    </TextContent>
                                </StackItem>

                                {
                                    rule.kbase_node_id &&
                                    <StackItem>
                                        <TextContent>
                                            <Text
                                                className={'icon-with-label'}
                                                component={TextVariants.a}
                                            >
                                                <a
                                                    href={`${RH_KB_URL}/${rule.kbase_node_id}`}
                                                    target="_blank"
                                                    rel="noopener noreferrer"
                                                >
                                                    {
                                                        <FormattedMessage {...messages.knowledgebaseArticle}/>
                                                    }
                                                    <ExternalLinkAltIcon className="l-sm-spacer"/>
                                                </a>
                                            </Text>
                                        </TextContent>
                                    </StackItem>
                                }

                            </Stack>
                        </GridItem>
                        <GridItem span={4}>
                            <Stack gutter="md">
                                {rule.rule_id &&
                                    <StackItem>
                                        <TextContent>
                                            <Text
                                                component={TextVariants.a}
                                                onClick={() => handleExposedSystemFilter(rule.rule_id)}
                                            >
                                                <Link
                                                    key={rule.rule_id}
                                                    to={`/cves/${synopsis}/?security_rule=${rule.rule_id}`}
                                                >
                                                    <FormattedMessage {...messages.filterByAffectedSystems}/>
                                                </Link>

                                            </Text>
                                        </TextContent>
                                    </StackItem>
                                }
                                <StackItem>
                                    <TextContent>
                                        <Text component={TextVariants.h4}>
                                            <FormattedMessage {...messages.remediatioLabel}/>
                                        </Text>
                                        <Split gutter="md">
                                            <SplitItem>
                                                <Label className="label">
                                                    <FormattedMessage {...messages.ansibleRemediation}/>
                                                </Label>
                                            </SplitItem>
                                            <SplitItem className="icon-with-label" isFilled>
                                                { !rule.playbook_count ? <FormattedMessage {...messages.no}/>
                                                    : (
                                                        <Fragment>
                                                            <CheckCircleIcon className="ansible-success" />
                                                            <FormattedMessage {...messages.yes}/>
                                                            <Tooltip
                                                                content={
                                                                    <FormattedMessage {...messages.ansibleRemediationTooltip}/>
                                                                }
                                                            >
                                                                <OutlinedQuestionCircleIcon
                                                                    color={'var(--pf-global--secondary-color--100)'}
                                                                    className="l-sm-spacer"
                                                                />
                                                            </Tooltip>
                                                        </Fragment>
                                                    )

                                                }
                                            </SplitItem>
                                        </Split>
                                        <Split gutter="md">
                                            <SplitItem>
                                                <Label className="label"><FormattedMessage {...messages.riskOfChange}/></Label>
                                            </SplitItem>
                                            <SplitItem isFilled className="icon-with-label">
                                                { RISK_OF_CHANGE_LABEL[rule.change_risk] }

                                                <Tooltip content={RISK_OF_CHANGE_TOOLTIP[rule.change_risk]}>
                                                    <OutlinedQuestionCircleIcon
                                                        color={'var(--pf-global--secondary-color--100)'}
                                                        className="l-sm-spacer"
                                                    />
                                                </Tooltip>

                                                <br/>
                                                {
                                                    rule.reboot_required && <strong><Reboot red /></strong>
                                                }
                                            </SplitItem>
                                        </Split>
                                    </TextContent>
                                </StackItem>
                            </Stack>
                        </GridItem>
                    </Grid>
                </CardBody>
            </Card>
        ))}
    </Fragment>;
};

CSAwRuleBox.defaultProps = {
    rules: []
};

CSAwRuleBox.propTypes = {
    rules: PropTypes.array,
    synopsis: PropTypes.string,
    fetchAffectedSystems: PropTypes.func
};

export default CSAwRuleBox;
