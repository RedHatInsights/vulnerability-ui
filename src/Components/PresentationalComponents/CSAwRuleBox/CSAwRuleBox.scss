.label {
    min-width: 140px;
}

.l-sm-spacer {
    margin-left: var(--pf-global--spacer--sm);
}

.ansible-success {
    color: var(--pf-global--success-color--200);
}


.card-box {
    border: 1px solid var(--pf-global--BorderColor--dark-100);
    box-shadow: none;
    padding: var(--pf-global--spacer--sm) !important;

    .pf-c-expandable-section__content {
        margin-top: 0;
    }

    .rule-card-body {
        padding-left: 44px;
        padding-top: var(--pf-global--spacer--sm);
    }

    // align toggle '>' icon with title text
    .pf-c-expandable-section__toggle-icon {
        margin: 2px 5px 0px 8px;
    }
    
    // vertically align 'Filter by X exposed systems' label
    #filter-affected-systems-split {
        display: inline-block;
    }

    #filter-affected-systems {
        display: inline-block;
        vertical-align: middle;
    }
}

.fontsize--lg {
    font-size: var(--pf-global--FontSize--lg);
}

.checkCircleIcon {
    color: var(--pf-global--success-color--100);
    vertical-align: -0.15em;
}

.powerOffIcon {
    color: var(--pf-global--danger-color--100);
}

.outlinedQuestionCircleIcon {
    color: var(--pf-global--secondary-color--100)
}
