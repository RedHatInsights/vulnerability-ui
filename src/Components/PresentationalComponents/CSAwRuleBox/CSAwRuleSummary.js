/* eslint-disable react/no-danger */
import React from 'react';
import marked from 'marked';
import PropTypes from 'prop-types';
import { injectIntl } from 'react-intl';
import sanitizeHtml from 'sanitize-html';
import { Truncate } from '@redhat-cloud-services/frontend-components';
import { Stack, StackItem, TextContent } from '@patternfly/react-core';
import { handleCVELink } from '../../../Helpers/VulnerabilityHelper';
import { TRUNCATE_TEXT_THRESHOLD } from '../../../Helpers/constants';
import messages from '../../../Messages';

/**
 * extend marked's renderer in order to open link in new tab
 */
let renderer = new marked.Renderer();
renderer.link = function() {
    let link = marked.Renderer.prototype.link.apply(this, arguments);
    return link.replace('<a', '<a target=\'_blank\'');
};

marked.setOptions({ renderer });

const CSAwRuleSummary = ({ text, truncate, link, intl }) => {
    const dangerousHtml = (text) => ({ __html: sanitizeHtml(text) });

    return (
        <StackItem>
            <TextContent className="rule-description">
                { truncate && text.length > TRUNCATE_TEXT_THRESHOLD
                    ? (
                        <Truncate
                            length={TRUNCATE_TEXT_THRESHOLD}
                            expandText={intl.formatMessage(messages.readMore)}
                            collapseText={intl.formatMessage(messages.readLess)}
                            text={marked(text)}
                            spaceBetween
                        />
                    ) : (
                        <Stack>
                            <StackItem>
                                <span dangerouslySetInnerHTML={dangerousHtml(marked(text))}/>
                            </StackItem>
                            {link && <StackItem className="rule-link pf-u-mt-sm">
                                {handleCVELink(link, intl.formatMessage(messages.readMore))}
                            </StackItem>}
                        </Stack>
                    )
                }
            </TextContent>
        </StackItem>
    );

};

CSAwRuleSummary.defaultProps = {
    truncate: true,
    link: null
};

CSAwRuleSummary.propTypes = {
    intl: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.object
    ]),
    truncate: PropTypes.bool,
    text: PropTypes.string.isRequired,
    link: PropTypes.string
};

export default injectIntl(CSAwRuleSummary);
