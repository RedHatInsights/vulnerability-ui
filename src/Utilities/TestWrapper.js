import propTypes from 'prop-types';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { IntlProvider } from '@redhat-cloud-services/frontend-components-translations';

const TestWrapper = ({ children, store }) => {
    return (
        <IntlProvider locale="en">
            { store ?
                <Provider store={store}>
                    <Router>
                        {children}
                    </Router>
                </Provider> : children
            }
        </IntlProvider>
    );
};

TestWrapper.propTypes = {
    children: propTypes.node,
    store: propTypes.object
};

export default TestWrapper;
