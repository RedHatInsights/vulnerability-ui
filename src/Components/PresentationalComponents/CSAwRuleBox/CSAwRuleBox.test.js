import CSAwRuleBox from './CSAwRuleBox'
import { BrowserRouter as Router } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { initialState } from '../../../Store/Reducers/CVEDetailsPageStore';
import { render, screen } from '@testing-library/react';
import { IntlProvider } from '@redhat-cloud-services/frontend-components-translations';
import userEvent from '@testing-library/user-event';
import TestWrapper from '../../../Utilities/TestWrapper';

jest.mock('../../../Store/Actions/Actions', () => ({
    ...jest.requireActual('../../../Store/Actions/Actions'),
    changeExposedDevicesParameters: jest.fn(),
    changeExposedSystemsParameters: jest.fn()
}));

const mockStore = configureStore([store => next => action => {}]);
let store = mockStore(initialState);

beforeEach(() => {
    store.clearActions();
});

const rules = [
    {
        "associated_cves": [
            "CVE-2019-11135"
        ],
        "change_risk": 3,
        "description": "CVE-2019-11135: \"TAA\" kernel side-channel",
        "generic": "An implementation issue similar to MDS was found in many x86 microprocessors.\n\nAn unprivileged attacker can use this flaw to bypass memory restrictions in order to read cached data from the processor and sibling logical processors. This allows malicious users to access information from the local system, and allows malicious virtual machines or users thereof to access information from other guests or the host system.\n",
        "kbase_node_id": 4572391,
        "playbook_count": 1,
        "reboot_required": false,
        "rule_id": "CVE_2019_11135_cpu_taa|CVE_2019_11135_CPU_TAA_KERNEL",
        "summary": "An issue was found in the manner in which certain x86 microprocessors manage thread memory access synchronization. It has been assigned [CVE-2019-11135](https://access.redhat.com/security/cve/CVE-2019-11135). Complete mitigations for this issue require microcode and kernel updates, and disabling hyperthreading where applicable.\n\nAn unprivileged user can use this flaw to gain read access to privileged memory that would otherwise be inacessible, including from other virtual machines on the same host system.\n",
        "rule_impact": 2
    }
]
const synopsis = 'CVE-2019-11135';
const setHeaderFilters = jest.fn();

describe('CSAW Rule Box', () => {
    it('Should render with rules', () => {
        const { asFragment } = render(
            <TestWrapper store={ store }>
                <CSAwRuleBox rules={rules} synopsis={synopsis} setHeaderFilters={setHeaderFilters}/>
            </TestWrapper>
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it('Should render without reboot warning', () => {
        const rules = [
            {
                "associated_cves": [
                    "CVE-2019-11135"
                ],
                "change_risk": 3,
                "description": "CVE-2019-11135: \"TAA\" kernel side-channel",
                "generic": "An implementation issue similar to MDS was found in many x86 microprocessors.\n\nAn unprivileged attacker can use this flaw to bypass memory restrictions in order to read cached data from the processor and sibling logical processors. This allows malicious users to access information from the local system, and allows malicious virtual machines or users thereof to access information from other guests or the host system.\n",
                "kbase_node_id": 4572391,
                "playbook_count": 1,
                "reboot_required": false,
                "rule_id": "CVE_2019_11135_cpu_taa|CVE_2019_11135_CPU_TAA_KERNEL",
                "summary": "An issue was found in the manner in which certain x86 microprocessors manage thread memory access synchronization. It has been assigned [CVE-2019-11135](https://access.redhat.com/security/cve/CVE-2019-11135). Complete mitigations for this issue require microcode and kernel updates, and disabling hyperthreading where applicable.\n\nAn unprivileged user can use this flaw to gain read access to privileged memory that would otherwise be inacessible, including from other virtual machines on the same host system.\n"
            }
        ]

        render(
            <TestWrapper store={ store }>
                <CSAwRuleBox rules={rules} synopsis={synopsis} setHeaderFilters={setHeaderFilters}/>
            </TestWrapper>
        );

        expect(screen.queryByTestId('power-off-icon')).toBeNull();
    });

    it('Should render with reboot warning', () => {
        const rules = [
            {
                "associated_cves": [
                    "CVE-2019-11135"
                ],
                "change_risk": 3,
                "description": "CVE-2019-11135: \"TAA\" kernel side-channel",
                "generic": "An implementation issue similar to MDS was found in many x86 microprocessors.\n\nAn unprivileged attacker can use this flaw to bypass memory restrictions in order to read cached data from the processor and sibling logical processors. This allows malicious users to access information from the local system, and allows malicious virtual machines or users thereof to access information from other guests or the host system.\n",
                "kbase_node_id": 4572391,
                "playbook_count": 1,
                "reboot_required": true,
                "rule_id": "CVE_2019_11135_cpu_taa|CVE_2019_11135_CPU_TAA_KERNEL",
                "summary": "An issue was found in the manner in which certain x86 microprocessors manage thread memory access synchronization. It has been assigned [CVE-2019-11135](https://access.redhat.com/security/cve/CVE-2019-11135). Complete mitigations for this issue require microcode and kernel updates, and disabling hyperthreading where applicable.\n\nAn unprivileged user can use this flaw to gain read access to privileged memory that would otherwise be inacessible, including from other virtual machines on the same host system.\n"
            }
        ];
        
        render(
            <TestWrapper store={ store }>
                <CSAwRuleBox rules={rules} synopsis={synopsis} setHeaderFilters={setHeaderFilters}/>
            </TestWrapper>
        );

        expect(screen.getByTestId('power-off-icon')).not.toBeNull();
    });

    it('Should render assosiated cve links', () => {
        const rules = [
            {
                "associated_cves": [
                    "CVE-2019-11135",
                    "CVE-2019-11132",
                    "CVE-2019-11133"
                ],
                "change_risk": 3,
                "description": "CVE-2019-11135: \"TAA\" kernel side-channel",
                "generic": "An implementation issue similar to MDS was found in many x86 microprocessors.\n\nAn unprivileged attacker can use this flaw to bypass memory restrictions in order to read cached data from the processor and sibling logical processors. This allows malicious users to access information from the local system, and allows malicious virtual machines or users thereof to access information from other guests or the host system.\n",
                "kbase_node_id": 4572391,
                "playbook_count": 1,
                "reboot_required": true,
                "rule_id": "CVE_2019_11135_cpu_taa|CVE_2019_11135_CPU_TAA_KERNEL",
                "summary": "An issue was found in the manner in which certain x86 microprocessors manage thread memory access synchronization. It has been assigned [CVE-2019-11135](https://access.redhat.com/security/cve/CVE-2019-11135). Complete mitigations for this issue require microcode and kernel updates, and disabling hyperthreading where applicable.\n\nAn unprivileged user can use this flaw to gain read access to privileged memory that would otherwise be inacessible, including from other virtual machines on the same host system.\n"
            }
        ]

        render(
            <TestWrapper store={ store }>
                <CSAwRuleBox rules={rules} synopsis={synopsis} setHeaderFilters={setHeaderFilters}/>
            </TestWrapper>
        );

        expect(screen.getAllByTestId('associated-cve-link')).toHaveLength(2);
    });

    it('Should render with playbook', () => {
        const rules = [
            {
                "associated_cves": [
                    "CVE-2019-11135",
                    "CVE-2019-11132",
                    "CVE-2019-11133"
                ],
                "change_risk": 3,
                "description": "CVE-2019-11135: \"TAA\" kernel side-channel",
                "generic": "An implementation issue similar to MDS was found in many x86 microprocessors.\n\nAn unprivileged attacker can use this flaw to bypass memory restrictions in order to read cached data from the processor and sibling logical processors. This allows malicious users to access information from the local system, and allows malicious virtual machines or users thereof to access information from other guests or the host system.\n",
                "kbase_node_id": 4572391,
                "playbook_count": 1,
                "reboot_required": true,
                "rule_id": "CVE_2019_11135_cpu_taa|CVE_2019_11135_CPU_TAA_KERNEL",
                "summary": "An issue was found in the manner in which certain x86 microprocessors manage thread memory access synchronization. It has been assigned [CVE-2019-11135](https://access.redhat.com/security/cve/CVE-2019-11135). Complete mitigations for this issue require microcode and kernel updates, and disabling hyperthreading where applicable.\n\nAn unprivileged user can use this flaw to gain read access to privileged memory that would otherwise be inacessible, including from other virtual machines on the same host system.\n"
            }
        ];

        render(
            <TestWrapper store={ store }>
                <CSAwRuleBox rules={rules} synopsis={synopsis} setHeaderFilters={setHeaderFilters}/>
            </TestWrapper>
        );

        expect(screen.getAllByTestId('check-circle-icon')).toHaveLength(1);
    });

    it('Should render without playbook', () => {
        const rules = [
            {
                "associated_cves": [
                    "CVE-2019-11135",
                    "CVE-2019-11132",
                    "CVE-2019-11133"
                ],
                "change_risk": 3,
                "description": "CVE-2019-11135: \"TAA\" kernel side-channel",
                "generic": "An implementation issue similar to MDS was found in many x86 microprocessors.\n\nAn unprivileged attacker can use this flaw to bypass memory restrictions in order to read cached data from the processor and sibling logical processors. This allows malicious users to access information from the local system, and allows malicious virtual machines or users thereof to access information from other guests or the host system.\n",
                "kbase_node_id": 4572391,
                "playbook_count": 0,
                "reboot_required": true,
                "rule_id": "CVE_2019_11135_cpu_taa|CVE_2019_11135_CPU_TAA_KERNEL",
                "summary": "An issue was found in the manner in which certain x86 microprocessors manage thread memory access synchronization. It has been assigned [CVE-2019-11135](https://access.redhat.com/security/cve/CVE-2019-11135). Complete mitigations for this issue require microcode and kernel updates, and disabling hyperthreading where applicable.\n\nAn unprivileged user can use this flaw to gain read access to privileged memory that would otherwise be inacessible, including from other virtual machines on the same host system.\n"
            }
        ];

        render(
            <TestWrapper store={ store }>
                <CSAwRuleBox rules={rules} synopsis={synopsis} setHeaderFilters={setHeaderFilters}/>
            </TestWrapper>
        );

        expect(screen.queryByTestId('check-circle-icon')).toBeNull();
    });

    it('Should render with knowledge base link', () => {
        const rules = [
            {
                "associated_cves": [
                    "CVE-2019-11135",
                    "CVE-2019-11132",
                    "CVE-2019-11133"
                ],
                "change_risk": 3,
                "description": "CVE-2019-11135: \"TAA\" kernel side-channel",
                "generic": "An implementation issue similar to MDS was found in many x86 microprocessors.\n\nAn unprivileged attacker can use this flaw to bypass memory restrictions in order to read cached data from the processor and sibling logical processors. This allows malicious users to access information from the local system, and allows malicious virtual machines or users thereof to access information from other guests or the host system.\n",
                "kbase_node_id": 4572391,
                "playbook_count": 0,
                "reboot_required": true,
                "rule_id": "CVE_2019_11135_cpu_taa|CVE_2019_11135_CPU_TAA_KERNEL",
                "summary": "An issue was found in the manner in which certain x86 microprocessors manage thread memory access synchronization. It has been assigned [CVE-2019-11135](https://access.redhat.com/security/cve/CVE-2019-11135). Complete mitigations for this issue require microcode and kernel updates, and disabling hyperthreading where applicable.\n\nAn unprivileged user can use this flaw to gain read access to privileged memory that would otherwise be inacessible, including from other virtual machines on the same host system.\n"
            }
        ];

        render(
            <TestWrapper store={ store }>
                <CSAwRuleBox rules={rules} synopsis={synopsis} setHeaderFilters={setHeaderFilters}/>
            </TestWrapper>
        );

        expect(screen.getAllByTestId('kb-link')).toHaveLength(1);
    });

    it('Should render without knowledge base link', () => {
        const rules = [
            {
                "associated_cves": [
                    "CVE-2019-11135",
                    "CVE-2019-11132",
                    "CVE-2019-11133"
                ],
                "change_risk": 3,
                "description": "CVE-2019-11135: \"TAA\" kernel side-channel",
                "generic": "An implementation issue similar to MDS was found in many x86 microprocessors.\n\nAn unprivileged attacker can use this flaw to bypass memory restrictions in order to read cached data from the processor and sibling logical processors. This allows malicious users to access information from the local system, and allows malicious virtual machines or users thereof to access information from other guests or the host system.\n",
                "kbase_node_id": null,
                "playbook_count": 0,
                "reboot_required": true,
                "rule_id": "CVE_2019_11135_cpu_taa|CVE_2019_11135_CPU_TAA_KERNEL",
                "summary": "An issue was found in the manner in which certain x86 microprocessors manage thread memory access synchronization. It has been assigned [CVE-2019-11135](https://access.redhat.com/security/cve/CVE-2019-11135). Complete mitigations for this issue require microcode and kernel updates, and disabling hyperthreading where applicable.\n\nAn unprivileged user can use this flaw to gain read access to privileged memory that would otherwise be inacessible, including from other virtual machines on the same host system.\n"
            }
        ];

        render(
            <TestWrapper store={ store }>
                <CSAwRuleBox rules={rules} synopsis={synopsis} setHeaderFilters={setHeaderFilters}/>
            </TestWrapper>
        );

        expect(screen.queryByTestId('kb-link')).toBeNull();
    });

    it('Should filter based on the error key', () => {
        const rules = [
            {
                "associated_cves": [
                    "CVE-2019-11135",
                    "CVE-2019-11132",
                    "CVE-2019-11133"
                ],
                "change_risk": 3,
                "description": "CVE-2019-11135: \"TAA\" kernel side-channel",
                "generic": "An implementation issue similar to MDS was found in many x86 microprocessors.\n\nAn unprivileged attacker can use this flaw to bypass memory restrictions in order to read cached data from the processor and sibling logical processors. This allows malicious users to access information from the local system, and allows malicious virtual machines or users thereof to access information from other guests or the host system.\n",
                "kbase_node_id": null,
                "playbook_count": 0,
                "reboot_required": true,
                "rule_id": "CVE_2019_11135_cpu_taa|CVE_2019_11135_CPU_TAA_KERNEL",
                "summary": "An issue was found in the manner in which certain x86 microprocessors manage thread memory access synchronization. It has been assigned [CVE-2019-11135](https://access.redhat.com/security/cve/CVE-2019-11135). Complete mitigations for this issue require microcode and kernel updates, and disabling hyperthreading where applicable.\n\nAn unprivileged user can use this flaw to gain read access to privileged memory that would otherwise be inacessible, including from other virtual machines on the same host system.\n"
            }
        ];

        render(
            <TestWrapper store={ store }>
                <CSAwRuleBox rules={rules} synopsis={synopsis} setHeaderFilters={setHeaderFilters}/>
            </TestWrapper>
        );

        userEvent.click(screen.getByTestId('filter-affected-systems'));

        expect(setHeaderFilters).toBeCalledTimes(1)
        expect(setHeaderFilters).toHaveBeenCalledWith({ rule: rules[0].rule_id })

    });
});
