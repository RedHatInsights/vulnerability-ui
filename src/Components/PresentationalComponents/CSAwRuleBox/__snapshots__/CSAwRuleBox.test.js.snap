// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CSAW Rule Box Should render with rules 1`] = `
<Fragment>
  <Card
    className="card-box-shadow"
    key="CVE_2019_11135_cpu_taa|CVE_2019_11135_CPU_TAA_KERNEL"
  >
    <CardBody>
      <Grid
        gutter="md"
      >
        <GridItem
          span={12}
        >
          <TextContent
            className="icon-with-label"
          >
            <Text
              component="h3"
            >
              <Tooltip
                appendTo={[Function]}
                aria="describedby"
                boundary="window"
                className=""
                content={
                  <FormattedMessage
                    defaultMessage="Denotes a security rule. Security rules are written by Red Hat to help you configure your systems"
                    description="Tooltip"
                    id="rules.iconTooolip"
                    values={Object {}}
                  />
                }
                distance={15}
                enableFlip={true}
                entryDelay={500}
                exitDelay={500}
                flipBehavior={
                  Array [
                    "top",
                    "right",
                    "bottom",
                    "left",
                    "top",
                    "right",
                    "bottom",
                  ]
                }
                id=""
                isAppLauncher={false}
                isContentLeftAligned={false}
                isVisible={false}
                maxWidth="18.75rem"
                position="top"
                tippyProps={Object {}}
                trigger="mouseenter focus"
                zIndex={9999}
              >
                <CSAwIcon />
              </Tooltip>
              CVE-2019-11135: "TAA" kernel side-channel
            </Text>
          </TextContent>
        </GridItem>
        <GridItem
          span={8}
        >
          <Stack
            gutter="md"
          >
            <StackItem>
              <TextContent>
                <TruncatedText
                  characterLimit={250}
                  component="p"
                  content="An issue was found in the manner in which certain x86 microprocessors manage thread memory access synchronization. It has been assigned [CVE-2019-11135](https://access.redhat.com/security/cve/CVE-2019-11135). Complete mitigations for this issue require microcode and kernel updates, and disabling hyperthreading where applicable.

An unprivileged user can use this flaw to gain read access to privileged memory that would otherwise be inacessible, including from other virtual machines on the same host system.
"
                  showLessLabel={
                    <FormattedMessage
                      defaultMessage="Read less"
                      description="Read less label for general usage"
                      id="readLess"
                      values={Object {}}
                    />
                  }
                  showMoreLabel={
                    <FormattedMessage
                      defaultMessage="Read More"
                      description="Read more generic label"
                      id="readMore"
                      values={Object {}}
                    />
                  }
                />
              </TextContent>
            </StackItem>
            <StackItem>
              <TextContent>
                <Text
                  component="h4"
                >
                   
                  <FormattedMessage
                    defaultMessage="Associated CVEs:"
                    description="Associated CVEs general label"
                    id="associatedCves"
                    values={Object {}}
                  />
                </Text>
                <Text
                  component="p"
                >
                  CVE-2019-11135
                   (
                  <FormattedMessage
                    defaultMessage="current"
                    description="Current label for general usage"
                    id="current"
                    values={Object {}}
                  />
                  )
                </Text>
              </TextContent>
            </StackItem>
            <StackItem>
              <TextContent>
                <Text
                  className="icon-with-label"
                  component="a"
                >
                  <a
                    href="https://access.redhat.com/node/4572391"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    <FormattedMessage
                      defaultMessage="Knowledgebase article"
                      description="Knowledgebase article"
                      id="knowledgebasearticle"
                      values={Object {}}
                    />
                    <ExternalLinkAltIcon
                      className="l-sm-spacer"
                      color="currentColor"
                      noVerticalAlign={false}
                      size="sm"
                      title={null}
                    />
                  </a>
                </Text>
              </TextContent>
            </StackItem>
          </Stack>
        </GridItem>
        <GridItem
          span={4}
        >
          <Stack
            gutter="md"
          >
            <StackItem>
              <TextContent>
                <Text
                  component="a"
                  onClick={[Function]}
                >
                  <Link
                    key="CVE_2019_11135_cpu_taa|CVE_2019_11135_CPU_TAA_KERNEL"
                    replace={false}
                    to="/cves/CVE-2019-11135/?security_rule=CVE_2019_11135_cpu_taa|CVE_2019_11135_CPU_TAA_KERNEL"
                  >
                    <FormattedMessage
                      defaultMessage="Filter by affected systems"
                      description="Temporaty csaw filter"
                      id="filterByAffectedSystems"
                      values={Object {}}
                    />
                  </Link>
                </Text>
              </TextContent>
            </StackItem>
            <StackItem>
              <TextContent>
                <Text
                  component="h4"
                >
                  <FormattedMessage
                    defaultMessage="Remediation"
                    description="Remediation label for general usage "
                    id="remediation"
                    values={Object {}}
                  />
                </Text>
                <Split
                  gutter="md"
                >
                  <SplitItem>
                    <Label
                      className="label"
                    >
                      <FormattedMessage
                        defaultMessage="Ansible remediation"
                        description="label for remediating with Ansible"
                        id="ansibleRemediation"
                        values={Object {}}
                      />
                    </Label>
                  </SplitItem>
                  <SplitItem
                    className="icon-with-label"
                    isFilled={true}
                  >
                    <CheckCircleIcon
                      className="ansible-success"
                      color="currentColor"
                      noVerticalAlign={false}
                      size="sm"
                      title={null}
                    />
                    <FormattedMessage
                      defaultMessage="Yes"
                      description="Yes general label"
                      id="yes"
                      values={Object {}}
                    />
                    <Tooltip
                      appendTo={[Function]}
                      aria="describedby"
                      boundary="window"
                      className=""
                      content={
                        <FormattedMessage
                          defaultMessage="You can create Ansible Playbooks and remediate your systems with Remediation application"
                          description="Tooltip for remediating with Ansible"
                          id="ansibleRemediationTooltip"
                          values={Object {}}
                        />
                      }
                      distance={15}
                      enableFlip={true}
                      entryDelay={500}
                      exitDelay={500}
                      flipBehavior={
                        Array [
                          "top",
                          "right",
                          "bottom",
                          "left",
                          "top",
                          "right",
                          "bottom",
                        ]
                      }
                      id=""
                      isAppLauncher={false}
                      isContentLeftAligned={false}
                      isVisible={false}
                      maxWidth="18.75rem"
                      position="top"
                      tippyProps={Object {}}
                      trigger="mouseenter focus"
                      zIndex={9999}
                    >
                      <OutlinedQuestionCircleIcon
                        className="l-sm-spacer"
                        color="var(--pf-global--secondary-color--100)"
                        noVerticalAlign={false}
                        size="sm"
                        title={null}
                      />
                    </Tooltip>
                  </SplitItem>
                </Split>
                <Split
                  gutter="md"
                >
                  <SplitItem>
                    <Label
                      className="label"
                    >
                      <FormattedMessage
                        defaultMessage="Risk of change"
                        description="Risk of change label for CVE rules"
                        id="riskOfChange"
                        values={Object {}}
                      />
                    </Label>
                  </SplitItem>
                  <SplitItem
                    className="icon-with-label"
                    isFilled={true}
                  >
                    <ExclamationTriangleIcon
                      color="var(--pf-global--warning-color--100)"
                      noVerticalAlign={false}
                      size="sm"
                      title={null}
                    />
                    Moderate
                    <Tooltip
                      appendTo={[Function]}
                      aria="describedby"
                      boundary="window"
                      className=""
                      content="These will likely require an outage window."
                      distance={15}
                      enableFlip={true}
                      entryDelay={500}
                      exitDelay={500}
                      flipBehavior={
                        Array [
                          "top",
                          "right",
                          "bottom",
                          "left",
                          "top",
                          "right",
                          "bottom",
                        ]
                      }
                      id=""
                      isAppLauncher={false}
                      isContentLeftAligned={false}
                      isVisible={false}
                      maxWidth="18.75rem"
                      position="top"
                      tippyProps={Object {}}
                      trigger="mouseenter focus"
                      zIndex={9999}
                    >
                      <OutlinedQuestionCircleIcon
                        className="l-sm-spacer"
                        color="var(--pf-global--secondary-color--100)"
                        noVerticalAlign={false}
                        size="sm"
                        title={null}
                      />
                    </Tooltip>
                    <br />
                  </SplitItem>
                </Split>
              </TextContent>
            </StackItem>
          </Stack>
        </GridItem>
      </Grid>
    </CardBody>
  </Card>
</Fragment>
`;

exports[`CSAW Rule Box Should render without rules  1`] = `<Fragment />`;
